src/backend/joanie/signature/backends/lex_persona/client.py

```diff
@@ -0,0 +1,414 @@
+"""Base Signature with Lex Persona Backend"""
+import json
+import os
+import re
+import time
+from logging import getLogger
+from typing import Dict, List
+
+import requests
+from rest_framework import status
+from rest_framework.response import Response
+
+from joanie.core.utils import webhooks
+from joanie.signature import exceptions
+
+from ..base import BaseSignatureBackend
+
+logger = getLogger(__name__)
+
+
+# pylint: disable=too-many-instance-attributes
+class LexPersonaClient(BaseSignatureBackend):
+    """
+    The Signature base class.
+    It contains generic methods to trigger
+    the workflow process of signing a document,
+    and to retrieve it once it has been signed.
+    """
+
+    name = "lex_persona"
+
+    def __init__(self, configuration):
+        """
+        Access data must be kept secret.
+        Replace those values in your .env file with your
+        own credentials provided by Lex Persona.
+        """
+        super().__init__(configuration)
+        self._set_configuration_attributes(configuration)
+        created, validity = self._set_create_and_validity_date()
+        self.workflow_id = str()
+        self.create_payload = {
+            "name": "Contract Definition / Convention de formation",
+            "description": "A simple document that engages the student to follow the course",
```

Comment:
use the fields from ContractDefinition?

Reply:
I have updated the following way for now :

```
"name": str(self.contract.definition.title),
"description": "Academic Engagement",
```
It does not seem that we have a description field on `Contract`, or `Contract Definition` or `Order`.

I have decided to add "Academic Engagement" has the description instead. 

The description is only used by the signature provider at only one place, which is the page that preceeds the signature of the document.

Since we do not meet this page anymore thank to the method  `get_signature_invitation_link`, the user who has to sign the document will never see this description.